hal/network/ncp_client/sara/sara_ncp_client.cpp

```diff
@@ -966,6 +967,126 @@ int SaraNcpClient::checkNetConfForImsi() {
+    bool reset = false;
+    int resetCount = 0;
+    bool disableLowPowerModes = false;
+    // Note: Not failing on AT error on ICCID/IMSI lookup since SIMs have shown strange edge cases
+    // where they error for no reason, and hard resetting the modem or power cycling would not clear it.
+    //
+    // Check if we are using a Twilio Super SIM based on ICCID
+    char buf[32] = {};
+    int ccidCount = 0;
+    do {
+        memset(buf, 0, sizeof(buf));
+        auto lenCcid = getIccidImpl(buf, sizeof(buf));
+        if (lenCcid > 5) {
+            netConf_ = networkConfigForIccid(buf, lenCcid);
+            break;
+        }
+    } while (++ccidCount < CCID_MAX_RETRY_CNT);
+    // If failed above i.e., netConf_ is still not valid, look for network settings based on IMSI
+    if (!netConf_.isValid()) {
+        CHECK(checkNetConfForImsi());
+    }
+
+    do {
+        // Set UMNOPROF and UBANDMASK as appropriate based on SIM
+        auto respUmnoprof = parser_.sendCommand("AT+UMNOPROF?");
+        reset = false;
+        int curProf = static_cast<int>(UbloxSaraUmnoprof::NONE);
+        auto r = CHECK_PARSER(respUmnoprof.scanf("+UMNOPROF: %d", &curProf));
+        CHECK_PARSER_OK(respUmnoprof.readResult());
+        // First time setup, or switching between official SIM on wrong profile?
+        if (r == 1 && (static_cast<UbloxSaraUmnoprof>(curProf) == UbloxSaraUmnoprof::SW_DEFAULT ||
+                (netConf_.netProv() == CellularNetworkProvider::TWILIO &&
+                static_cast<UbloxSaraUmnoprof>(curProf) != UbloxSaraUmnoprof::STANDARD_EUROPE) ||
+                (netConf_.netProv() == CellularNetworkProvider::KORE_ATT &&
+                static_cast<UbloxSaraUmnoprof>(curProf) != UbloxSaraUmnoprof::ATT)) ) {
+            int newProf = static_cast<int>(UbloxSaraUmnoprof::SIM_SELECT);
+
+            // Hard code ATT for 05.12 firmware versions
+            if (netConf_.netProv() == CellularNetworkProvider::KORE_ATT) {
+                if (getAppFirmwareVersion() == UBLOX_NCP_R4_APP_FW_VERSION_0512) {
+                    newProf = static_cast<int>(UbloxSaraUmnoprof::ATT);
+                }
+            }
+            // TWILIO Super SIM
+            if (netConf_.netProv() == CellularNetworkProvider::TWILIO) {
+                // _oldFirmwarePresent: u-blox firmware 05.06* and 05.07* does not have
+                // UMNOPROF=100 available. Default to UMNOPROF=0 in that case.
+                if (oldFirmwarePresent_) {
+                    if (static_cast<UbloxSaraUmnoprof>(curProf) == UbloxSaraUmnoprof::SW_DEFAULT) {
+                        break;
+                    } else {
+                        newProf = static_cast<int>(UbloxSaraUmnoprof::SW_DEFAULT);
+                    }
+                } else {
+                    newProf = static_cast<int>(UbloxSaraUmnoprof::STANDARD_EUROPE);
+                }
+            }
+            // KORE AT&T or 3rd Party SIM
+            else {
```

Comment:
@keeramis Should this apply to the 05.12 workaround as well?

Reply:
Correct. Great catch. thanks! Noticed that 2 tries is an optimal number for ATT too. Added